src/protagonist/Orchestrator/Infrastructure/IIIF/IIIFCanvasFactory.cs

```diff
@@ -142,21 +163,21 @@ public class IIIFCanvasFactory
                     On = canvasId,
                     Resource = new IIIF2.ImageResource
                     {
-                        Id = GetFullQualifiedImagePath(i, customerPathElement,
+                        Id = GetFullQualifiedImagePath(asset, customerPathElement,
                             thumbnailSizes.MaxDerivativeSize, false),
                         Width = thumbnailSizes.MaxDerivativeSize.Width,
                         Height = thumbnailSizes.MaxDerivativeSize.Height,
-                        Service = GetImageServices(i, customerPathElement, null)
+                        Service = GetImageServices(asset, customerPathElement, null)
                     },
                 }.AsList()
             };
 
-            if (!thumbnailSizes.OpenThumbnails.IsNullOrEmpty())
+            if (!asset.HasDeliveryChannel(AssetDeliveryChannels.Thumbnails) || !thumbnailSizes.OpenThumbnails.IsNullOrEmpty())

```

Comment:
Is this logic correct? Should that be `if (asset.HasDeliveryChannel(AssetDeliveryChannels.Thumbnails)`? (ticket notes to only add thumbs if asset has thumbs channel)